### PR TITLE
CATROID-334 NO_VARIABLE _SELECTED should be translated

### PR DESCRIPTION
**Pull request by ameyalaad**
Issue Resolved.

### CODE REVIEW

**Comment by ameyalaad** [comment]
Hey @wslany , just to be clear, when the variable is empty, nothing needs to be done right?

(Sorry for the delay, had my exams on)

**Comment by wslany** [comment]
Hi @ameyalaad thanks a lot!! I checked the functionality, and it looks good to me now. Next step will be another round of code review. Once you are ready, please move it to "code review" on jira. All the best and Merry Christmas, Wolfgang

**Comment by ameyalaad** [comment]
Hey
I am unable to move the issue from the Kanban board in the Catrobat Dev Board. 
Also, could you put me into contact with someone who I could ask for help with further issues? That would be a huge help. Thanks and a Merry Christmas to you too!

**Comment by wslany** [comment]
> Hey
> I am unable to move the issue from the Kanban board in the Catrobat Dev Board.
> Also, could you put me into contact with someone who I could ask for help with further issues? That would be a huge help. Thanks and a Merry Christmas to you too!

Thanks --- I'll let you know soon. Normally I would just move it into the code review phase, but we are currently trying to get new members on-board in a more gentle way, so let me try that with you, which first may take a bit longer but in the long run will make things hopefully much better. In particular, it would be great, once you are on Slack, to ask for a remote one-on-one code review session, as that allows to solve issues immediately, instead of communicating over many days or even weeks synchronously via github. 

One more thing: It seems that your PR for CATROID-334 also fixes https://jira.catrob.at/browse/CATROID-333 on which @swlz is already working. Susanne, please can you comment on that. Maybe you can solve it together, since @ameyalaad 's PR is not yet code reviewed as well.

**Comment by wslany** [comment]
@ameyalaad just one more comment, preempting the code review --- I just looked now at your code, since as a PO I'm normally not responsible for the code quality: the PR is still missing automatic tests that check the correct new behavior. These will be needed before the PR can be accepted, as we must make sure that the functionality will never be lost in the future, Also, the tests serve as a documentation for other developers what your changes are doing, and for future maintainability. And, normally tests must be written *before* fixing bugs --- not afterwards. Check out test driven development, there's much more behind this.

File: catroid/src/main/java/org/catrobat/catroid/stage/ShowTextActor.java [hunk 1]
```diff
@@ -119,16 +120,14 @@ private void drawVariables(List<UserVariable> variableList, Batch batch) {
...
-			drawText(batch, Constants.NO_VARIABLE_SELECTED, xPosition, yPosition, color);
+			drawText(batch, Utils.getStringfromResource(R.string.no_variable_selected), xPosition, yPosition, color);
 		} else {
 			for (UserVariable variable : variableList) {
 				if (variable.getName().equals(variableToShow.getName())) {
 					String variableValue = variable.getValue().toString();
 					if (variable.getVisible()) {
 						if (isNumberAndInteger(variableValue)) {
 							drawText(batch, variableValueWithoutDecimal, xPosition, yPosition, color);
-						} else if (variableValue.isEmpty()) {
```
Comment (PlantinumX) [inline catroid/src/main/java/org/catrobat/catroid/stage/ShowTextActor.java]: please use this line and move line 159-162 there as discussed over slack

**Comment by PlantinumX** [comment]
Hi Ameya pls squash your commits to two commmits one for rrefactoring and one regarding the actual solve of the bug

File: catroid/src/main/java/org/catrobat/catroid/utils/Utils.java [hunk 2]
```diff
@@ -450,4 +451,8 @@ public static int getBit(int number, int index) {
 		}
 		return 0;
 	}
+
+	public static String getStringfromResource(int resourceId) {
+		return CatroidApplication.getAppContext().getString(resourceId);
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/utils/Utils.java]: accesssing context this way should be avoided.

this "getAppContext" method was a hack in the first place, put in place years ago, and we tried to remove it multiple times already.
please do not add further usages.

please find another way to put the string into its place, without accessing context from classes that did not need a context up to this point.

Comment (ameyalaad) [reply catroid/src/main/java/org/catrobat/catroid/utils/Utils.java]: Hey @84n4n4 , it seems that in order to access local application resources, android only provides the unique Resources class, which can be accessed only through the Context class. Under the hood of the resources class, it seems to use AssetManager (provided by the android system), but I am unable to get an instance of this class to get the string. Should I attempt to find a workaround with this AssetManager class?

**Comment by PlantinumX** [comment]
pls resolve conflicts and reopen pull request :)

**Pull request closed without merging**